src/compiler/array.rs

```diff
@@ -1005,3 +1005,143 @@ pub(crate) fn slice(
 
     Ok(new_arr)
 }
+
+fn copy_from_element(
+    compiler: &mut Compiler,
+    source_element: &Symbol,
+    target_data_ptr: &Symbol,
+    target_capacity: &Symbol,
+    element_width: u32,
+    index: u32,
+) -> Result<()> {
+    // Ensure that the target array has enough capacity to hold the source element
```

Comment:
As it's only used for unshift so it's basically an internal assertion, so it makes sense to have cfg(test) or similar?